main.go

```diff
@@ -306,14 +306,15 @@ func initializeManager(config *managerConfig) (manager.Manager, error) {
 func main() {
 	logger := setupLogger()
 
-	restConfig := ctrl.GetConfigOrDie()
-
-	config := &managerConfig{
-		logger:     logger,
-		restConfig: restConfig,
-	}
+	config := &managerConfig{}
 	parseFlags(config)
 
+	// Flags must be parsed before calling GetConfigOrDie, because
+	// it reads the value of the--kubeconfig flag.
+	config.restConfig = ctrl.GetConfigOrDie()
+
+	config.logger = logger
```

Comment:
I suspect that this assignment is a bug. I think we want the manager to use the logger we initialize in  `parseFlags()`.